packages/service-integration/src/service_integration/cli/_escaping.py

```diff
@@ -0,0 +1,44 @@
+import re
+from pathlib import Path
+from typing import Annotated
+
+import typer
+
+from ..osparc_config import OSPARC_CONFIG_DIRNAME
+
+
+def escape_dollar_brace(text: str) -> str:
+    # the pattern finds '$${' that is not preceded by another '$'.
+    pattern = r"(?<!\$)\$\${"
+    replacement = "$$$${"
+    return re.sub(pattern, replacement, text)
+
+
+def legacy_escape(
+    osparc_config_dirname: Annotated[
+        Path,
+        typer.Option(
+            "--osparc-config-dirname",
+            help="Path to where the .osparc configuration directory is located",
+        ),
+    ] = Path(OSPARC_CONFIG_DIRNAME),
+):
+    """
+    Replaces all '$${' sequences with '$$$${' unless they are part of an
+    existing '$$$${' sequence.
+    """
+
+    # NOTE: since https://github.com/docker/compose/releases/tag/v2.35.0
+    # docker-compose was fixed/changed to escape `${}`
+    # SEE https://github.com/docker/compose/pull/12664
```

Comment:
does that mean it will be fixed in 2.35.0? will this code still work then?

Reply:
From what I understood this is the new standard. I have to run a few more tests to see exactly in which version of docker all this broke.

Reply:
Tried to figure out when this started but could not point my finger on it. Will stop doing so. This will help us fix the issue in the build pipelines.